presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/ApplyTableScanRedirection.java

```diff
@@ -0,0 +1,169 @@
+        Optional<TableHandle> destinationTableHandle = metadata.getTableHandle(
+                context.getSession(),
+                convertFromSchemaTableName(destinationTable.getCatalogName()).apply(destinationTable.getSchemaTableName()));
+        if (destinationTableHandle.isEmpty()) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, format("Destination table %s from table scan redirection not found", destinationTable));
+        }
+        if (destinationTableHandle.get().equals(scanNode.getTable())) {
+            return Result.empty();
+        }
+
+        Map<ColumnHandle, String> columnMapping = tableScanRedirectApplicationResult.get().getDestinationColumns();
+        Map<String, ColumnHandle> destinationColumnHandles = metadata.getColumnHandles(context.getSession(), destinationTableHandle.get());
+        Map<Symbol, ColumnHandle> newAssignments = scanNode.getAssignments().entrySet().stream()
+                .collect(toImmutableMap(Map.Entry::getKey, entry -> {
+                    String destinationColumn = requireNonNull(
+                            columnMapping.get(entry.getValue()),
+                            format("Did not find mapping for source column %s in table scan redirection", entry.getValue()));
+                    return requireNonNull(
+                            destinationColumnHandles.get(destinationColumn),
+                            format("Did not find handle for column %s in destination table %s", destinationColumn, destinationTable));
+                }));
+
+        TupleDomain<String> enforcedConstraint = tableScanRedirectApplicationResult.get().getFilter();
+        if (enforcedConstraint.isAll()) {
+            return Result.ofPlanNode(
+                    new TableScanNode(
+                            scanNode.getId(),
+                            destinationTableHandle.get(),
+                            scanNode.getOutputSymbols(),
+                            newAssignments,
+                            TupleDomain.all(),
+                            scanNode.isForDelete()));
+        }
+
+        ImmutableMap.Builder<Symbol, ColumnHandle> newAssignmentsBuilder = ImmutableMap.<Symbol, ColumnHandle>builder()
+                .putAll(newAssignments);
+        ImmutableList.Builder<Symbol> newOutputSymbolsBuilder = ImmutableList.<Symbol>builder()
+                .addAll(scanNode.getOutputSymbols());
+        Map<ColumnHandle, Symbol> inverseAssignments = ImmutableBiMap.copyOf(scanNode.getAssignments()).inverse();
+        Map<String, ColumnHandle> inverseColumnsMapping = ImmutableBiMap.copyOf(columnMapping).inverse();
+        TupleDomain<Symbol> transformedConstraint = enforcedConstraint.transform(destinationColumn -> {
+            ColumnHandle sourceColumnHandle = requireNonNull(
+                    inverseColumnsMapping.get(destinationColumn),
+                    format("Did not find mapping for destination column %s in table scan redirection", destinationColumn));
```

Comment:
This should throw some kind of Presto exception. Otherwise, it will produce an INTERNAL_ERROR, when it's an issue with a missing or stale mapping within the connector.

Reply:
Updated the rule to throw COLUMN_NOT_FOUND or TABLE_NOT_FOUND PrestoException